lib/Sema/CSSimplify.cpp

```diff
@@ -1410,6 +1410,35 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
     case ConstraintKind::ArgumentConversion:
     case ConstraintKind::OperatorArgumentTupleConversion:
     case ConstraintKind::OperatorArgumentConversion:
+      if (type1->is<InOutType>() && !type2->is<InOutType>() && typeVar2) {
```

Comment:
`typeVar2` will only be non-null when `type2` is a type variable, so the `!type2->is<InOutType>()` check here is redundant.